src/main/java/cn/nukkit/block/BlockFireSoul.java

```diff
@@ -0,0 +1,36 @@
+package cn.nukkit.block;
+
+import cn.nukkit.api.PowerNukkitOnly;
+import cn.nukkit.api.Since;
+import cn.nukkit.utils.BlockColor;
+
+@PowerNukkitOnly
+@Since("1.4.0.0-PN")
+public class BlockFireSoul extends BlockFire {
```

Comment:
Please add override onUpdate to make it revert to normal fire in normal updates if the block bellow is not valid for soul fires
`getCurrentState().withBlockId(BlockID.FIRE).getBlock()`